includes/amp-helper-functions.php

```diff
@@ -384,13 +384,32 @@ function amp_get_slug() {
  * @return string Current URL.
  */
 function amp_get_current_url() {
-	$url = preg_replace( '#(^https?://[^/]+)/.*#', '$1', home_url( '/' ) );
+	$parsed_url = wp_parse_url( home_url() );
```

Comment:
How will this be handled if the URL failed to parse?

For example, if `home_url()` returned `:`, `wp_parse_url()` would return false.

Reply:
I suppose then we'd need this?

```suggestion
	$parsed_url = wp_parse_url( home_url() );
	if ( ! is_array( $parsed_url ) ) {
		$parsed_url = [];
	}
```

Reply:
Yea that should do it :+1: .

Reply:
Done in dd00644b20373dc374e034f460a3b2a7f32ecb2a